timesketch/lib/utils.py

```diff
@@ -68,6 +70,48 @@ def read_and_validate_csv(path):
             yield row
 
 
+def read_and_validate_jsonl(path):
+    """Generator for reading a JSONL (json lines) file.
+
+    Args:
+        path: Path to the JSONL file
+    """
+    # Fields that must be present in each entry of the JSONL file.
+    mandatory_fields = [u'message', u'datetime', u'timestamp_desc']
+
+    with open(path, 'rb') as fh:
+
+        lineno = 0
+        for line in fh:
+            lineno += 1
+            try:
+                linedict = json.loads(line)
+                if u'datetime' not in linedict.keys()\
+                  and u'timestamp' in linedict.keys():
+                    epoch = int(str(linedict[u'timestamp'])[:10])
```

Comment:
If you use the imported dateutil parser you can keep the precision. Also, see the refactor suggestion above.

Reply:
Please see comment above.

Reply:
Ack. SG